eth_bridge/ic/src/eth_proxy/src/common/types.rs

```diff
@@ -91,7 +94,7 @@ pub struct StableProxyState {
     /// store incoming messages against status locks
     pub incoming_messages: HashMap<MessageHash, MessageStatus>,
     /// user balances
-    pub balances: HashMap<Principal, HashMap<Principal, Nat>>,
+    pub balances: Option<HashMap<Principal, Vec<(Principal, Nat)>>>,
```

Comment:
```suggestion
pub balances: RefCell<HashMap<EthereumAddr, Nat>>,
```
The same usage can be applied here

Reply:
Also Is `Optional` being committed or only required for upgrades?

Reply:
it is required for the upgrades